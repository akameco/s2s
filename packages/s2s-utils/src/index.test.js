// @flow
import path from 'path'
import generate from 'babel-generator'
import * as utils from '.'

test('getImportPath same folder', () => {
  const result = utils.getImportPath('path/to/index.js', 'path/to/test.js')
  expect(result).toBe('./test')
})

test('getImportPath same folder when not ext', () => {
  const result = utils.getImportPath('path/to/index', 'path/to/test')
  expect(result).toBe('./test')
})

test('getImportPath parent folder', () => {
  const result = utils.getImportPath('path/to/nest/index.js', 'path/to/test.js')
  expect(result).toBe('../test')
})

test('template', () => {
  const ast = utils.template(`type Action = A`)()
  expect(ast).toMatchSnapshot()
})

test('inheritsOpts', () => {
  const parserOpts = { plugins: [] }
  utils.inheritsOpts().manipulateOptions({}, parserOpts)
  expect(parserOpts).toStrictEqual({ plugins: ['flow', 'objectRestSpread'] })
})

test('return parent path name', () => {
  const filePath = path.sep === '\\' ? 'ok\\hello\\world' : 'ok/hello/world'
  expect(utils.getParentDirName(filePath)).toBe('hello')
})

testAst([
  {
    title: 'snapshot createImportDeclaration when string array',
    ast: utils.createImportDeclaration(['local1', 'local2'], './source'),
  },
  {
    title: 'snapshot createImportDeclaration when string',
    ast: utils.createImportDeclaration('local', './source'),
  },
  {
    title: 'snapshot typeImport',
    ast: utils.typeImport('local', 'imported', 'source'),
  },
  {
    title: 'snapshot defaultImport',
    ast: utils.defaultImport('defaultlocal', 'source'),
  },
])

function testAst(tests: Array<{ title: string, ast: * }>) {
  for (const { title, ast } of tests) {
    test(title, () => {
      const { code } = generate(ast)
      expect(code).toMatchSnapshot(title)
    })
  }
}
