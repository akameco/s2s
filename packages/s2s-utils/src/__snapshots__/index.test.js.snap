// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot createImportDeclaration when string array: snapshot createImportDeclaration when string array 1`] = `"import { local1, local2 } from \\"./source\\";"`;

exports[`snapshot createImportDeclaration when string: snapshot createImportDeclaration when string 1`] = `"import { local } from \\"./source\\";"`;

exports[`snapshot defaultImport: snapshot defaultImport 1`] = `"import defaultlocal from \\"source\\";"`;

exports[`snapshot typeImport: snapshot typeImport 1`] = `"import type { imported as local } from \\"source\\";"`;

exports[`template 1`] = `
Node {
  "__clone": undefined,
  "_fromTemplate": true,
  "end": undefined,
  "id": Node {
    "__clone": undefined,
    "_fromTemplate": true,
    "end": undefined,
    "loc": undefined,
    "name": "Action",
    "start": undefined,
    "type": "Identifier",
  },
  "loc": undefined,
  "right": Node {
    "__clone": undefined,
    "_fromTemplate": true,
    "end": undefined,
    "id": Node {
      "__clone": undefined,
      "_fromTemplate": true,
      "end": undefined,
      "loc": undefined,
      "name": "A",
      "start": undefined,
      "type": "Identifier",
    },
    "loc": undefined,
    "start": undefined,
    "type": "GenericTypeAnnotation",
    "typeParameters": null,
  },
  "start": undefined,
  "type": "TypeAlias",
  "typeParameters": null,
}
`;
